Doc/using/cmdline.rst

```diff
@@ -994,3 +1000,110 @@ Debug-mode variables
    Need Python configured with the :option:`--with-trace-refs` build option.
 
    .. versionadded:: 3.11
+
+.. _using-on-finding-modules:
+
+Finding modules
+---------------
+
+A module search path is initialized when Python starts. This module search path
+may be accessed at :data:`sys.path`.
+
+The first entry in the module search path is the directory that contains the
+input script, if there is one. Otherwise, the first entry is the current
+directory, which is the case when executing the interactive shell, a :ref:`-c <using-on-interface-option-c>`
+command, or :ref:`-m <using-on-interface-option-m>` module.
+
+The :envvar:`PYTHONPATH` environment variable is often used to add directories
+to the search path. If this environment variable is found then the contents are
+added to the module search path.
+
+.. note::
+
+   :envvar:`PYTHONPATH` will affect all installed Python versions/environments.
+   Be wary of setting this in your shell profile or global environment variables.
+   The :mod:`site` module offers more nuanced techniques as mentioned below.
+
+The next items added are the directories containing standard Python modules as
+well as any :term:`extension module`\s that these modules depend on. Extension
+modules are .pyd files on Windows and .so files on other platforms. The
+directory with the platform-independent Python modules is called ``prefix``.
+The directory with the extension modules is called ``exec_prefix``.
+
+The :envvar:`PYTHONHOME` environment variable may be used to set the ``prefix``
+and ``exec_prefix`` locations. Otherwise these directories are found by using
+the Python executable as a starting point and then looking for various 'landmark'
+files and directories. Note that any symbolic links are followed so the real
+Python executable location is used as the search starting point. The Python
+executable location is called ``home``.
+
+Once ``home`` is determined, the ``prefix`` directory is found by first looking
+for :file:`python{majorversion}{minorversion}.zip` (``python311.zip``). On Windows
+the zip archive is searched for in ``home`` and on Unix the archive is expected
+to be in :file:`lib`. Note that the expected zip archive location is added to the
+module search path even if the archive does not exist. If no archive was found,
+Python on Windows will continue the search for ``prefix`` by looking for :file:`Lib\\os.py`.
+Python on Unix will look for :file:`lib/python{majorversion}.{minorversion}/os.py`
+(``lib/python3.11/os.py``). On Windows ``prefix`` and ``exec_prefix`` are the same,
+however on other platforms :file:`lib/python{majorversion}.{minorversion}/lib-dynload`
+(``lib/python3.11/lib-dynload``) is searched for and used as an anchor for
+``exec_prefix``.
```

Comment:
You should document sys.platlibdir and PYTHONPLATLIBDIR env var somewhere. 

See: https://docs.python.org/dev/library/sys.html#sys.platlibdir

Reply:
Thanks, I have now mentioned them.